dali/test/python/test_operator_coord_transform.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+from test_utils import check_batch, dali_type
+
+def make_param(kind, shape):
+    if kind == "input":
+        return fn.uniform(range=(0, 1), shape=shape)
+    elif kind == "scalar input":
+        return fn.reshape(fn.uniform(range=(0, 1)), shape=[])
+    elif kind == "vector":
+        return np.random.rand(*shape).astype(np.float32)
+    elif kind == "scalar":
+        return np.random.rand()
+    else:
+        return None
+
+def clip(value, type = None):
+    try:
+        info = np.iinfo(type)
+        return np.clip(value, info.min, info.max)
+    except:
+        return value
+
+def make_data_batch(batch_size, in_dim, type):
+    np.random.seed(1234)
+    batch = []
+    lo = 0
+    hi = 1
+    if np.issubdtype(type, np.integer):
+        info = np.iinfo(type)
+        # clip range to +/- 1000000 to prevent excessively large epsilons
+        lo = max(info.min / 2, -1000000)
+        hi = min(info.max / 2, 1000000)
+
+    for i in range(batch_size):
+        batch.append((np.random.rand(np.random.randint(1, 10000), in_dim)*(hi-lo) + lo).astype(type))
+    return batch
+
+def get_data_source(batch_size, in_dim, type):
+    return lambda: make_data_batch(batch_size, in_dim, type)
+
+def _run_test(device, batch_size, out_dim, in_dim, in_dtype, out_dtype, M_kind, T_kind):
+    pipe = dali.pipeline.Pipeline(batch_size=batch_size, num_threads=2, device_id=0, seed=1234)
+    with pipe:
+        X = fn.external_source(source=get_data_source(batch_size, in_dim, in_dtype), device=device)
+        M = make_param(M_kind, [out_dim, in_dim])
+        T = make_param(T_kind, [out_dim])
+        Y = fn.coord_transform(X,
```

Comment:
Aren't you supposed to set the `device` in `coord_transform` as well?

Reply:
That's the beauty of it :) The wrapper function checks the inputs - for GPU DataNodes - if it finds any, GPU backend is automatically used.

Reply:
Is that `beaty` documented anywhere?

Reply:
Hmm... good question. I think I wrote it _somewhere_, but I can't find it now. We should add it to the `functional_api.rst`, I guess. I don't want to touch RST files while the docs PR is in review.

Reply:
OK